cmd/imagescan.go

```diff
@@ -1,28 +1,36 @@
 package cmd
 
 import (
+	"fmt"
+
 	"github.com/accuknox/accuknox-cli-v2/pkg/imagescan"
 	kubesheildConfig "github.com/accuknox/kubeshield/pkg/scanner/config"
 	"github.com/spf13/cobra"
 )
 
 var (
-	HOST_NAME string
-	RUN_TIME  string
-	cfg       = kubesheildConfig.Config{}
+	HOST_NAME                   string
+	RUN_TIME                    string
+	artifactEndpointPath        string
+	cfg                         = kubesheildConfig.Config{}
+	defaultArtifactEndpointPath = "/api/v1/artifact/"
 )
 
 var imageScanCmd = &cobra.Command{
 	Use:   "image-scan",
 	Short: "scans vm container images",
-	Long: `Scans VM container images using trivy 
-and sends back the result to saas through artifact API
+	Long: `Scans VM container images 
+and sends back the result to saas
 		`,
 	Args: cobra.NoArgs,
-	PreRunE: func(cmd *cobra.Command, args []string) error {
-		return imagescan.IsTrivyInstalled()
-	},
 	RunE: func(cmd *cobra.Command, args []string) error {
+		if !imagescan.IsValidDomain(cfg.ScanConfig.ArtifactAPI) {
+			return fmt.Errorf("invalid domain name: %s", cfg.ScanConfig.ArtifactAPI)
+		}
+		if artifactEndpointPath == "" {
+			artifactEndpointPath = defaultArtifactEndpointPath
+		}
+		cfg.ScanConfig.ArtifactAPI = fmt.Sprintf("https://%s%s", cfg.ScanConfig.ArtifactAPI, artifactEndpointPath)
```

Comment:
can we sanitise this api url ?
lets say if user uses --artifactEndpointPath="this/endpoint/path" then the api will fail right?